### PR TITLE
boards: m5stack_core2: Add possibility to control 5V_BUS signal

### LINKED ISSUE #67277: boards: m5stack_core2: 5V bus is not controllable
**Describe the bug**
The m5stack_core2 board features a 5v bus domain, that for example is used by the grove connector. This 5V bus is not being supplied accordingly. In addition there currently is no possibility to enable/disable this power domain.

**To Reproduce**
- The external Gove connector 5V pin (2) is showing 0V.
- The documentation is not describing any behaviour on the 5V bus (https://docs.zephyrproject.org/latest/boards/xtensa/m5stack_core2/doc/index.html)
- device list is not showing any regulator to enable the 5V bus.
```
uart:~$ device list
devices:
- rtc@3ff48000 (READY)
- gpio@3ff44800 (READY)
- gpio@3ff44000 (READY)
- trng@3ff75144 (READY)
- uart@3ff40000 (READY)
- i2c@3ff53000 (READY)
- spi@3ff65000 (READY)
- axp192@34 (READY)
- axp192_gpio (READY)
- LDO3 (READY)
- LDO2 (READY)
- DCDC3 (READY)
- DCDC1 (READY)
- ili9342c@0 (READY)
- ft5336@38 (READY)
- lvgl_pointer (READY)
uart:~$ 
```

**Expected behavior**
The 5V may not be disabled by default, but there should be at least the possibility

**Impact**
Grove extension port is not usable properly right now.

**Logs and console output**
none

**Environment (please complete the following information):**
 - OS: MacOS
 - Toolchain Zephyr SDK 0.16.4
 - Comit SHA: 0a9728f87e41e2956cfd47de97c8f57e41b13aaa

**Additional context**
none

### PR DESCRIPTION
**Pull request by MrMarteng**
This commit fixes [#67277](https://github.com/zephyrproject-rtos/zephyr/issues/67277) and hence adds possibilityto properly utilize grove extension port on M5Stack Core2 module.

5V_BUS can be enabled via regulator bus_5v. By default this regulator is disabled.

### CODE REVIEW

**Comment by MrMarteng** [comment]
@kartben Do you maybe find some time to test it with some of your grove extension hardware?

File: boards/xtensa/m5stack_core2/m5stack_core2.dts [hunk 1]
```diff
@@ -125,6 +125,7 @@
 	axp192_pmic: axp192@34 {
 		compatible = "x-powers,axp192";
 		reg = <0x34>;
+		/delete-node/ vbusen-disable; /* make sure manual control VBUS is enabled */
```
Comment (gmarull) [inline boards/xtensa/m5stack_core2/m5stack_core2.dts:128]: why is this needed? where is this boolean property set?

Comment (MrMarteng) [reply boards/xtensa/m5stack_core2/m5stack_core2.dts:128]: I wanted to explicitly make sure, this option is false. That isn't required?

Comment (gmarull) [reply boards/xtensa/m5stack_core2/m5stack_core2.dts:128]: no, it's not

Comment (MrMarteng) [reply boards/xtensa/m5stack_core2/m5stack_core2.dts]: Fixed in https://github.com/zephyrproject-rtos/zephyr/commit/17544d699aa18a4226001f50cab33aad23c8d225

**Comment by MrMarteng** [comment]
@gmarull @aasinclair Could you please have a look at this PR?

File: drivers/mfd/mfd_axp192.c [hunk 2]
```diff
@@ -164,9 +189,11 @@ int mfd_axp192_gpio_func_get(const struct device *dev, uint8_t gpio, enum axp192
 		return -EINVAL;
 	}
 
-	ret = i2c_reg_read_byte_dt(&(config->i2c), gpio_reg_desc[gpio].reg, &reg_fnc);
-	if (ret != 0) {
-		return ret;
+	if (gpio < ARRAY_SIZE(gpio_reg_desc)) {
```
Comment (aasinclair) [inline drivers/mfd/mfd_axp192.c:192]: Is it worth moving the special case for EXTEN here instead?
It can return immediately.

File: drivers/mfd/mfd_axp192.c [hunk 3]
```diff
@@ -355,18 +387,25 @@ int mfd_axp192_gpio_func_ctrl(const struct device *dev, const struct device *cli
...
 	if (ret != 0) {
 		LOG_ERR("Invalid function (0x%x) for gpio %d", func, gpio);
 		return ret;
 	}
 
-	ret = i2c_reg_update_byte_dt(&(config->i2c), gpio_reg_desc[gpio].reg,
-				     gpio_reg_desc[gpio].mask, reg_cfg);
-	if (ret != 0) {
-		return ret;
+	if (gpio < ARRAY_SIZE(gpio_reg_desc)) {
```
Comment (aasinclair) [inline drivers/mfd/mfd_axp192.c:403]: Could you return earlier if the user asks for an invalid channel?

Comment (MrMarteng) [reply drivers/mfd/mfd_axp192.c:403]: That's not possible here. The write operation can only be executed after the switch-case above has been evaluated. The if is needed to filter the new EXT_EN gpio, that doesn't require a configuration to be written.

File: drivers/mfd/mfd_axp192.c (hunk 2 above)
Comment (MrMarteng) [reply drivers/mfd/mfd_axp192.c:192]: I am not sure what you mean here?

Are you proposing adding a separate if for setting `*func` for the EXTEN case (5) ?

**Review by aasinclair — COMMENTED** [review]
Needs a rebase, but otherwise looks look

**Comment by MrMarteng** [comment]
Branch has been rebased

**Review by aasinclair — APPROVED** [review]

**Review by gmarull — APPROVED** [review]
